### ISSUE TITLE
`isStandardSyntaxRule` trusts the `extend` mark `postcss-less` reads out of a quoted attribute value

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#171]
Found while fixing [#151](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/151), which teaches `isStandardSyntaxSelector` to read past a quoted attribute value. One reader never reaches that util, and answers the same question wrongly on its own.

### What minimal example or steps are needed to reproduce the bug?

```less
[title=":extend(x)"] {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-less",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/string-quotes": "single"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

`Expected single quotes`, and `--fix` writing `[title=':extend(x)']` — the same answer the rule gives for a `.css` file holding that selector, since the text inside the quotes is an attribute value and not a Less `:extend`.

### What actually happened?

Nothing is reported, and nothing is fixed. `postcss-less` marks the rule:

```js
less.parse(`[title=":extend(x)"] {}`).first.extend        // true
less.parse(`[data-x="a:extend(y)"] {}`).first.extend      // true
```

and `isStandardSyntaxRule` reads that mark before it asks anything else:

```js
// Ignore Less &:extend rule
if (`extend` in rule && rule.extend) return false
```

so every rule built on it — `no-extra-semicolons`, `selector-attribute-brackets-space-inside`, `selector-descendant-combinator-no-non-space`, `selector-list-comma-newline-after`, `selector-pseudo-class-case`, `selector-pseudo-class-parentheses-space-inside`, `selector-pseudo-element-case` and `string-quotes` — passes the rule node over. Only a `.less` file is affected; the same selector in a `.css` or `.scss` file is checked and fixed.

### Do you have a proposal to fix the bug?

The mark comes from the text of the selector, quotes and all, so it is wrong for the same reason #151 was, and `isStandardSyntaxSelector` already knows how to answer it: it blanks the quoted runs before it reads anything.

Two ways out, and the second is the honest one.

- Ask the util as well: `if (rule.extend && !isStandardSyntaxSelector(rule.selector)) return false`. One line, and it leans on the util already reading `:extend` past a quoted value.
- Report it to `postcss-less`, since the mark itself is what is wrong, and a `.extend` that answers "is this an extend rule" ought to answer it about the code rather than about the text. Whatever comes of that, the line above keeps this plugin right in the meantime.

Worth checking either way that a real `&:extend(.b)` and a rule carrying both, `[title=":extend(x)"]:extend(.b)`, still read as Less.
